spec.bs

```diff
@@ -810,6 +810,83 @@ To <dfn>compute protocol matches a special scheme flag</dfn> given a [=construct
   1. If the result of running [=protocol component matches a special scheme=] given |protocol component| is true, then set |parser|'s [=constructor string parser/protocol matches a special scheme flag=] to true.
 </div>
 
+<h2 id=urlpatternlist-class>The {{URLPatternList}} class </h2>
+
+<xmp class="idl">
+[Exposed=(Window,Worker)]
+interface URLPatternList {
+  constructor(sequence<URLPatternListEntry> entries);
+
+  USVString? test(optional URLPatternInput input = {}, optional USVString baseURL);
+
+  URLPatternListResult? exec(optional URLPatternInput input = {}, optional USVString baseURL);
+};
+
+dictionary URLPatternListEntry {
+  required (USVString or URLPattern) pattern;
+  required USVString key;
+};
+
+dictionary URLPatternListResult {
+  URLPatternResult match;
+  USVString key;
+};
+</xmp>
+
+Note: The {{URLPatternList}} is a utility that combines many patterns into one
+matcher. This functionality can also be achieved in a naive userland
+implementation that iterates over a list of patterns and matches each one
+individually. This has the unfortunate effect of not being very fast. With the
+built-in {{URLPatternList}} implementers can use a radix tree for matching under
```

Comment:
Recommend just mentioning optimizations and not "radix tree" specifically.  That is just one kind of optimization.